### DISCUSSION TITLE
get compression algorithm of a dwarfs file

### DISCUSSION THREAD

**Discussion by cipitaua** [mhx/dwarfs#165] [Q&A] [answered]
in the past I have made some dwarfs compressed with `-l 9` and other with `-l 7`,  now I would like to recompress the 9 into the faster 7, how can I understand which is what?

**Comment by mhx** [comment, marked as answer]
You can, for example, use `dwarfsck`:
```
$ dwarfsck -d3 filtered.dwarfs
DwarFS version 2.4 [2]
SECTION num=0, type=BLOCK, compression=ZSTD, length=19,247, checksum=0x4eccf90e9bd127a3, offset=64, blocksize=80532, ratio=23.90%
SECTION num=1, type=METADATA_V2_SCHEMA, compression=ZSTD, length=405, checksum=0x0f9db3bcfcc64f17, offset=19375, blocksize=889, ratio=45.56%
SECTION num=2, type=METADATA_V2, compression=ZSTD, length=329, checksum=0x1edb6a0f438e223d, offset=19844, blocksize=334, ratio=98.50%
SECTION num=3, type=SECTION_INDEX, compression=NONE, length=32, checksum=0x8ce389cda5886088, offset=20237, blocksize=32, ratio=100.00%
created by: libdwarfs v0.6.2-13-ged609b8-dirty on branch wip
[...]
```
The `compression=` parts will tell you which compression was used for each section of the DwarFS image. Since the compression algorithm can be chosen independently of the level you specify with `-l` (and even independently for different section types), it's hard to tell if a file system image was built using `-l 9` or `-l 7`. However, as you're referring to "fast" vs "slow", you're probably referring to `zstd` vs `lzma`. So you'll have to check whether any `BLOCK` sections use `LZMA` compression; this will likely mean the image was built with `-l 9`.

**Reply by cipitaua** [reply]
crystal clear, thank you
